apps/indexer/src/utils/getImageSize.ts

```diff
@@ -0,0 +1,60 @@
+import { Readable } from "stream";
+import { ReadableStream } from "stream/web";
+import { imageSize } from "image-size";
+
+/**
+ * lightweight function to get the dimension of an image from a response,
+ * it stops reading the stream immediately after getting the dimension
+ * @param response
+ * @returns
+ */
+export async function getImageSize(
+  response: Response,
+  abortController: AbortController,
+): Promise<{ width: number; height: number } | undefined> {
+  if (!response.ok || !response.body) {
+    return;
+  }
+
+  const contentLength = parseInt(response.headers.get("content-length") ?? "0");
+
+  if (isNaN(contentLength) || contentLength <= 0) {
+    return;
+  }
+
+  const stream = Readable.fromWeb(response.body as unknown as ReadableStream);
+
+  return new Promise((resolve, reject) => {
+    const buffer = Buffer.alloc(contentLength, 0);
```

Comment:
we need to find out what's the common length for the image size metadata to appear and limit it to that if contentLength > commonLength. this prevent people deliberately response with large content-length to cause our server to use up all memory.

also remember to test what happen when chunk size exceed total buffer size when doing `buffer.write(chunk)`

Reply:
fixed